app/views/registrations/edit.html.erb

```diff
@@ -3,90 +3,45 @@
-        <%= f.label :username, class: 'col-sm-2 control-label' %>
-        <div class="col-sm-10"><%= f.text_field :username %></div>
-      </div>
-
-      <div class="field-wrapper">
-        <%= f.label :email, class: 'col-sm-2 control-label' %>
-        <div class="col-sm-10"><%= f.email_field :email %></div>
-      </div>
-
-      <div class="field-wrapper">
-        <%= f.label :locale, class: 'col-sm-2 control-label' %>
-        <div class="col-sm-10"><%= f.select :locale, I18n.available_locales, include_blank: true %></div>
-      </div>
-
-      <div class="field-wrapper">
-        <%= f.label :time_zone, class: 'col-sm-2 control-label' %>
-        <div class="col-sm-10"><%= f.time_zone_select :time_zone, nil, default: Time.zone %></div>
-      </div>
+  <div class="col-xs-12 col-md-8 col-md-offset-2">
+    <div class="page-header">
+      <h4 class="page-header-title darkgrey-2">
+        <i class="mi md-20">account_circle</i> <%= t('edit') %> <%= resource.class.name.humanize %>
+      </h4>
+    </div>
+  </div>
 
-      <div class="field-wrapper">
-        <%= f.label :password, class: 'col-sm-2 control-label' %>
-        <div class="col-sm-10">
-          <%= f.password_field :password, autocomplete: "off" %>
-          <div class="help-block"><%= t('.leave_blank_if_you_dont_want_to_change_it') %></div>
+  <div class="col-xs-12 col-md-5 col-md-offset-2">
+    <div class="panel panel-default card">
+      <div class="panel-body">
+        <div class="edit-user">
+          <%= render 'form' %>
         </div>
       </div>
+    </div>
+  </div>
 
-      <div class="field-wrapper">
-        <%= f.label :password_confirmation, autocomplete: "off", class: 'col-sm-2 control-label' %>
-        <div class="col-sm-10"><%= f.password_field :password_confirmation %></div>
-      </div>
-
-      <h3><%= t('.notify_me_when_my_stories_are') %></h3>
-
-      <div class="checkbox">
-        <%= f.check_box :email_delivery %>
-        <%= f.label :email_delivery, t('story.state.delivered') %>
-      </div>
-
-      <div class="checkbox">
-        <%= f.check_box :email_acceptance %>
-        <%= f.label :email_acceptance, t('story.state.accepted') %>
-      </div>
-
-      <div class="checkbox">
-        <%= f.check_box :email_rejection %>
-        <%= f.label :email_rejection, t('story.state.rejected') %>
-      </div>
-
-      <div class="field-wrapper current-password">
-        <%= f.label :current_password, class: 'col-sm-2 control-label' %>
-        <div class="col-sm-10">
-          <%= f.password_field :current_password, autocomplete: "off" %>
-          <div class="help-block">
-            <%= t('.we_need_your_current_password_to_confirm_your_changes') %>
-          </div>
+  <div class="col-xs-12 col-sm-3">
+    <div class="panel panel-default card user-profile">
+      <div class="panel-body">
+        <div class="text-center">
+          <% cache "user/#{:email}/avatar_edit_page", expires_in: 1.hour do %>
```

Comment:
Shouldn't this be `"user/#{@user.email}/avatar_edit_page"`? This will use always the same key for all users, which is wrong.